### PR TITLE
Added doctests, exceptions, type hints, fixed bug, remove trailing whitespace for infix_to_prefix_conversion

### PR DESCRIPTION
**Pull request by hollowcrust**
### Describe your change:

Fixes #9943 

File affected: data_structures/stacks/infix_to_prefix_conversion.py

- Added doctests
- Added exceptions for expressions with invalid bracket positions
- Added type hints for functions
- Edited code to ensure no trailing whitespaces in code and doctests
- Fixed a bug on line 53 (original file) by adding a condition `and stack[-1] != "("`

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Review by sanskark — APPROVED** [review]

File: data_structures/stacks/infix_to_prefix_conversion.py [hunk 1]
```diff
@@ -26,13 +26,13 @@ def infix_2_postfix(infix):
         "+": 1,
         "-": 1,
     }  # Priority of each operator
-    print_width = len(infix) if (len(infix) > 7) else 7
 
+    print_width = len(infix) if len(infix) > 7 else 7
```
Comment (cclauss) [inline data_structures/stacks/infix_to_prefix_conversion.py:30]:
```suggestion
    print_width = min(len(infix), 7)
```

File: data_structures/stacks/infix_to_prefix_conversion.py [hunk 2]
```diff
@@ -15,7 +15,7 @@
 """
 
 
-def infix_2_postfix(infix):
+def infix_2_postfix(infix: str) -> str:
```
Comment (cclauss) [inline data_structures/stacks/infix_to_prefix_conversion.py:18]: BEFORE we modify this function we need it to have doctests.  You can do that in a separate pull request and then we can circle back to this pull request AFTER the doctests have been merged.

Comment (hollowcrust) [reply data_structures/stacks/infix_to_prefix_conversion.py:18]: I'm reaching my PR limit so maybe I can commit changes directly here or can you help me review #10073 or #10012 first?

Comment (cclauss) [reply data_structures/stacks/infix_to_prefix_conversion.py:18]: I merged one of them that adds tests.  The other one deleted the work of others but perhaps it is the right thing to do.

Comment (hollowcrust) [reply data_structures/stacks/infix_to_prefix_conversion.py:18]: ok thanks i'll work on this

Comment (hollowcrust) [reply data_structures/stacks/infix_to_prefix_conversion.py:18]: @cclauss I added tests in #10095, you can help me look through.

**Comment by cclauss** [comment]
Please rebase to fix to resolve the git conflict.

**Comment by hollowcrust** [comment]
@cclauss I created a new PR #10259, you can help me check it please.
